src/screening/ScreeningFilters.jsx

```diff
@@ -1,50 +1,103 @@
-          placeholder="mock-ID-0"
-        />
-        <span>-</span>
-        <FieldInput />
-        <span>-</span>
-        <Dropdown
-          dropDownStyle={{
-            border: '1px solid black'
-          }}
-          placeholder="mock-ID-0"
-        />
-      </span>
-    </div>
-    <div className="name-filter-container">
-      <div>Name/id search:</div>
-      <span className="name-filter">
-        <Glyphicon className="left-icon" glyph="search" />
-        <FormControl
-          autoFocus
-          placeholder="search by text"
-          type="text"
-        />           
-        <Glyphicon className="right-icon" glyph="remove-sign" />
-      </span>
-    </div>
-  </div>
-);
+class ScreeningFilters extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleTextInputSearch = this.handleTextInputSearch.bind(this);
+
+    this.state = {
+      firstIdentifiers: new Set(),
+      thirdIdentifiers: new Set(),
+      firstIdentifierSearchValue: '',
+      secondIdentifierSearchValue: '',
+      thirdIdentifierSearchValue: ''
+    };
+  }
+
+  componentWillMount() {
+    if (this.props.rowData) {
+      this.props.rowData.map(data => {
+        return utils.getPatientIdentifiers(data) && utils.getPatientIdentifiers(data).split('<br/>').map(identifier =>
+          this.setState(({ firstIdentifiers, thirdIdentifiers }) => ({
+            firstIdentifiers: new Set(firstIdentifiers.add(identifier.split('-')[0])),
+            thirdIdentifiers: new Set(thirdIdentifiers.add(identifier.split('-')[2])),
+          })
+          ));
+      });
+    }
+  }
+
+  sanitizedIdentifiers = array => array.filter(function(el) { return el; });
+  handleUndefinedValues = (value, defaultValue) => typeof value === 'undefined' ? defaultValue : value;
+
+  handleSearch(field, value, location) {
+    const { firstIdentifierSearchValue, secondIdentifierSearchValue, thirdIdentifierSearchValue } = this.state;
+    let first, second, third, searchValue;
+
+    if (location === 'first') {
+      first = this.handleUndefinedValues(value, '');
+      this.setState({ firstIdentifierSearchValue : this.handleUndefinedValues(value, '') });
+      searchValue = `${first}${secondIdentifierSearchValue}${thirdIdentifierSearchValue}`;
+    } else if (location === 'second') {
+      second = this.handleUndefinedValues(value, '');
+      this.setState({ secondIdentifierSearchValue: this.handleUndefinedValues(value, '') });
+      searchValue = `${firstIdentifierSearchValue}${second}${thirdIdentifierSearchValue}`;
+    } else if (location === 'third') {
+      third = this.handleUndefinedValues(value, '');
+      this.setState({ thirdIdentifierSearchValue: this.handleUndefinedValues(value, '') });
+      searchValue = `${firstIdentifierSearchValue}${secondIdentifierSearchValue}${third}`;
+    }
+
+    this.props.handleSearchChange(searchValue, 'identifierFilter');
```

Comment:
We don't want to have the identifier filter change the search filter--they should be separate filters.